### PR TITLE
OPT - map sh to sf before tracking [requires dipy 1.8.0]

### PR DESCRIPTION
**Pull request by gabknight**
# Quick description

Mapping the SH to SF before tracking speed up det/prob by a factor 2X and PTT by 10X.

**requires dipy 1.8.0+**

## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

...

# Checklist

- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-782/1/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-782/2/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
@gabknight can you prepare this PR to be dipy 1.8.0 ready. I will merge it once dipy 1.8.0 is released.

**Comment by gabknight** [comment]
This PR is dipy 1.8.0 ready.

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-782/6/

**Comment by gabknight** [comment]
It should be all good. Let see if jenkins finds something.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scilpy/tracking/utils.py [hunk 1]
```diff
@@ -71,6 +71,11 @@ def add_tracking_options(p):
                          type=float, default=0.1,
                          help='Spherical function relative threshold. '
                               '[%(default)s]')
+    track_g.add_argument('--sh_to_pmf', dest='sh_to_pmf',
+                         type=bool, default=False,
```
[the remark this answers is not in the document]
Comment (gabknight) [reply scilpy/tracking/utils.py]: fixed. thanks @AntoineTheb

**Review by AntoineTheb — APPROVED** [review]
GTG @arnaudbore

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-782/9/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**